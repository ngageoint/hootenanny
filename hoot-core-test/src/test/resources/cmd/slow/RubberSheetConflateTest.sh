#!/bin/bash
set -e

# See ticket #6208 for an explanation

mkdir -p test-output/cmd/RubberSheetConflateTest
hoot conflate -C RemoveReview2Pre.conf -D osm2ogr.ops=hoot::DecomposeBuildingRelationsVisitor -D conflate.add.score.tags=yes -D services.db.writer.overwrite.map=true -D services.db.writer.create.user=true -D services.db.writer.email=test@test.com -D services.db.reader.email=test@test.com -D "map.cleaner.transforms=hoot::ReprojectToPlanarOp;hoot::DuplicateWayRemover;hoot::SuperfluousWayRemover;hoot::IntersectionSplitter;hoot::UnlikelyIntersectionRemover;hoot::DualWaySplitter;hoot::ImpliedDividedMarker;hoot::DuplicateNameRemover;hoot::SmallWayMerger;hoot::RemoveEmptyAreasVisitor;hoot::RemoveDuplicateAreaVisitor;hoot::NoInformationElementRemover;hoot::RubberSheet" -D "duplicate.name.remover.case.sensitive=true" -D "small.way.merger.threshold=15" -D "rubber.sheet.ref=false" -D "rubber.sheet.minimum.ties=10" -D "unify.optimizer.time.limit=30" -D "ogr.split.o2s=false" -D "ogr.tds.add.fcsubtype=true" -D "ogr.tds.structure=true" -D "conflate.enable.old.roads=true" -D "conflate.match.highway.classifier=hoot::HighwayExpertClassifier" -D "way.subline.matcher=hoot::MaximalSublineMatcher" -D "match.creators=hoot::BuildingMatchCreator;hoot::BuildingMatchCreator;hoot::ScriptMatchCreator,PoiGeneric.js" -D "merger.creators=hoot::BuildingMergerCreator;hoot::BuildingMergerCreator;hoot::ScriptMergerCreator" -D "search.radius.highway=-1" -D "way.matcher.heading.delta=5.0" -D "way.matcher.max.angle=60" -D "way.merger.min.split.size=5" -D "way.angle.sample.distance=10.0" -D "poi.polygon.match.name.threshold=0.8" -D "poi.polygon.match.review.distance=0" test-files/DcGisRoads.osm test-files/DcTigerRoads.osm test-output/cmd/RubberSheetConflateTest/output.osm
hoot --is-match test-output/cmd/RubberSheetConflateTest/output.osm test-files/cmd/slow/RubberSheetConflateTest/output.osm || diff test-output/cmd/RubberSheetConflateTest/output.osm test-files/cmd/slow/RubberSheetConflateTest/output.osm 

