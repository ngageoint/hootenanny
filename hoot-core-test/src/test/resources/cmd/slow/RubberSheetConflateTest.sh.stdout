15:29:19.976 INFO  ...pp/hoot/core/cmd/ConflateCmd.h( 123) bytesRead: 520169
15:29:20.187 INFO  ...pp/hoot/core/cmd/ConflateCmd.h( 136) inputBytes: 1.74787e+06
15:29:20.187 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::BuildingOutlineRemoveOp
15:29:20.187 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::MapCleaner
15:29:20.187 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::ReprojectToPlanarOp
Reprojecting 0 / 7487         Reprojecting 1000 / 7487         Reprojecting 2000 / 7487         Reprojecting 3000 / 7487         Reprojecting 4000 / 7487         Reprojecting 5000 / 7487         Reprojecting 6000 / 7487         Reprojecting 7000 / 7487         
15:29:20.248 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DuplicateWayRemover
15:29:20.248 INFO  ...nflate/DuplicateWayRemover.cpp(  61) Removing duplicate ways...
15:29:20.328 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SuperfluousWayRemover
15:29:20.328 INFO  ...late/SuperfluousWayRemover.cpp(  59) Removing superfluous ways...
15:29:20.328 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::IntersectionSplitter
15:29:20.328 INFO  ...itter/IntersectionSplitter.cpp( 109) Splitting intersections...
  Intersection splitter todo: 0       
15:29:20.351 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::UnlikelyIntersectionRemover
15:29:20.368 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DualWaySplitter
15:29:20.368 INFO  ...e/splitter/DualWaySplitter.cpp(  83) Assuming drives on right.
15:29:20.368 INFO  ...e/splitter/DualWaySplitter.cpp( 246) Splitting divided ways...

15:29:20.372 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::ImpliedDividedMarker
15:29:20.380 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DuplicateNameRemover
15:29:20.382 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SmallWayMerger
15:29:20.408 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  74) Applying visitor: hoot::RemoveEmptyAreasVisitor
15:29:20.428 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  74) Applying visitor: hoot::RemoveDuplicateAreaVisitor
15:29:20.451 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::NoInformationElementRemover
15:29:20.473 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::RubberSheet
15:29:20.650 INFO  .../core/conflate/RubberSheet.cpp( 325) Found 237 potential ties.
15:29:20.650 INFO  .../core/conflate/RubberSheet.cpp( 363) Found 173 confident ties.
15:29:20.670 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: DelaunayInterpolator RMSE: 0.9421
15:29:21.537 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: IDW, p: 0 RMSE: 0.759081
15:29:22.101 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: Kernel estimation, sigma: 3176.88 RMSE: 0.759046
15:29:22.101 INFO  .../core/conflate/RubberSheet.cpp( 258) Best candidate: Kernel estimation, sigma: 3176.88
15:29:22.101 INFO  .../core/conflate/RubberSheet.cpp( 372) Kernel estimation, sigma: 3176.88
15:29:22.672 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: Kernel estimation, sigma: 3177.26 RMSE: 0.75905
15:29:22.672 INFO  .../core/conflate/RubberSheet.cpp( 258) Best candidate: Kernel estimation, sigma: 3177.26
15:29:22.672 INFO  .../core/conflate/RubberSheet.cpp( 376) Kernel estimation, sigma: 3177.26
15:29:25.507 INFO  ...in/cpp/hoot/core/Conflator.cpp( 205) Preprocessing inputs...
15:29:25.508 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::ReprojectToPlanarOp
15:29:25.508 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DuplicateWayRemover
15:29:25.508 INFO  ...nflate/DuplicateWayRemover.cpp(  61) Removing duplicate ways...
15:29:25.606 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SuperfluousWayRemover
15:29:25.606 INFO  ...late/SuperfluousWayRemover.cpp(  59) Removing superfluous ways...
15:29:25.607 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::IntersectionSplitter
15:29:25.607 INFO  ...itter/IntersectionSplitter.cpp( 109) Splitting intersections...
  Intersection splitter todo: 0       
15:29:25.630 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::UnlikelyIntersectionRemover
15:29:25.641 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DualWaySplitter
15:29:25.641 INFO  ...e/splitter/DualWaySplitter.cpp(  83) Assuming drives on right.
15:29:25.641 INFO  ...e/splitter/DualWaySplitter.cpp( 246) Splitting divided ways...

15:29:25.646 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::ImpliedDividedMarker
15:29:25.654 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::DuplicateNameRemover
15:29:25.656 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SmallWayMerger
15:29:25.681 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  74) Applying visitor: hoot::RemoveEmptyAreasVisitor
15:29:25.702 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  74) Applying visitor: hoot::RemoveDuplicateAreaVisitor
15:29:25.725 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::NoInformationElementRemover
15:29:25.747 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::RubberSheet
15:29:25.910 INFO  .../core/conflate/RubberSheet.cpp( 325) Found 240 potential ties.
15:29:25.910 INFO  .../core/conflate/RubberSheet.cpp( 363) Found 176 confident ties.
15:29:25.928 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: DelaunayInterpolator RMSE: 0.865625
15:29:26.796 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: IDW, p: 0 RMSE: 0.752127
15:29:27.370 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: Kernel estimation, sigma: 3178.35 RMSE: 0.752132
15:29:27.371 INFO  .../core/conflate/RubberSheet.cpp( 258) Best candidate: IDW, p: 0
15:29:27.371 INFO  .../core/conflate/RubberSheet.cpp( 372) IDW, p: 0
15:29:28.243 INFO  .../core/conflate/RubberSheet.cpp( 245) candidate: IDW, p: 0 RMSE: 0.752127
15:29:28.243 INFO  .../core/conflate/RubberSheet.cpp( 258) Best candidate: IDW, p: 0
15:29:28.243 INFO  .../core/conflate/RubberSheet.cpp( 376) IDW, p: 0
15:29:30.788 INFO  ...in/cpp/hoot/core/Conflator.cpp( 229) Looking for manipulations...
15:29:30.788 INFO  ...ore/manipulators/WayMerger.cpp(  95) Finding all way merge manipulations...
  finding manipulations: 100 / 434          finding manipulations: 200 / 434          finding manipulations: 300 / 434          finding manipulations: 400 / 434        
15:29:35.016 INFO  ...in/cpp/hoot/core/Conflator.cpp( 231) Initial manipulation count: 789
15:29:35.016 INFO  ...in/cpp/hoot/core/Conflator.cpp( 112) Conflating map. Node count: 7500 way count: 1183
15:29:35.016 INFO  ...in/cpp/hoot/core/Conflator.cpp( 119) Manipulation heap 789
Current Score: 21.5902 it: 46 remaining: 828 elapsed: 1006ms             Current Score: 48.3319 it: 100 remaining: 871 elapsed: 2026ms             Current Score: 68.5004 it: 147 remaining: 885 elapsed: 3046ms             Current Score: 100.313 it: 227 remaining: 899 elapsed: 4058ms             Current Score: 130.644 it: 313 remaining: 911 elapsed: 5141ms             Current Score: 140.936 it: 361 remaining: 949 elapsed: 6160ms             Current Score: 150.66 it: 421 remaining: 958 elapsed: 7186ms             Current Score: 160.603 it: 582 remaining: 873 elapsed: 8246ms             Current Score: 169.291 it: 738 remaining: 853 elapsed: 9266ms             Current Score: 179.506 it: 833 remaining: 886 elapsed: 10280ms             Current Score: 186.284 it: 1115 remaining: 724 elapsed: 11299ms             Current Score: 190.814 it: 1811 remaining: 200 elapsed: 12302ms             
15:29:47.355 INFO  ...in/cpp/hoot/core/Conflator.cpp( 187) Conflating map. Node count: 7443 way count: 770
15:29:47.355 INFO  ...late/SuperfluousWayRemover.cpp(  59) Removing superfluous ways...
15:29:47.356 INFO  ...in/cpp/hoot/core/Conflator.cpp( 195) Conflating map. Node count: 7443 way count: 770
15:29:47.395 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  74) Applying visitor: hoot::RemoveElementsVisitor
15:29:47.398 INFO  ...lygon/BuildingMatchCreator.cpp( 200) *threshold: Thresholds; Match: 0.6 Miss: 0.6 Review: 0.6
15:29:47.399 INFO  ...lygon/BuildingMatchCreator.cpp( 236) Loading model from: /fouo/jason.surratt/hoot-release-3/hoot/conf/BuildingModel.rf
15:29:48.058 INFO  ...lygon/BuildingMatchCreator.cpp( 200) *threshold: Thresholds; Match: 0.6 Miss: 0.6 Review: 0.6
15:29:48.058 INFO  ...lygon/BuildingMatchCreator.cpp( 236) Loading model from: /fouo/jason.surratt/hoot-release-3/hoot/conf/BuildingModel.rf
15:29:48.854 INFO  ...conflate/UnifyingConflator.cpp( 165) Pre-constraining match count: 0
15:29:48.854 INFO  ...conflate/UnifyingConflator.cpp( 188) CM took: 1.90735e-06s.
15:29:48.854 INFO  ...conflate/UnifyingConflator.cpp( 189) CM Score: -1
15:29:48.854 INFO  ...conflate/UnifyingConflator.cpp( 197) GM took: 1.90735e-06s.
15:29:48.854 INFO  ...conflate/UnifyingConflator.cpp( 198) GM Score: -1
15:29:48.854 INFO  ...conflate/UnifyingConflator.cpp( 221) Post constraining match count: 0
15:29:48.854 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SuperfluousNodeRemover
15:29:48.854 INFO  ...ops/SuperfluousNodeRemover.cpp(  53) Removing superfluous nodes...
Reprojecting 0 / 7443         Reprojecting 1000 / 7443         Reprojecting 2000 / 7443         Reprojecting 3000 / 7443         Reprojecting 4000 / 7443         Reprojecting 5000 / 7443         Reprojecting 6000 / 7443         Reprojecting 7000 / 7443         
15:29:48.867 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::SmallWayMerger
15:29:48.883 INFO  .../cpp/hoot/core/ops/NamedOp.cpp(  56) Applying operation: hoot::BuildingOutlineUpdateOp
Reprojecting 0 / 7096         Reprojecting 1000 / 7096         Reprojecting 2000 / 7096         Reprojecting 3000 / 7096         Reprojecting 4000 / 7096         Reprojecting 5000 / 7096         Reprojecting 6000 / 7096         Reprojecting 7000 / 7096         
